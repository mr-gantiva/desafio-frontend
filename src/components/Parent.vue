<script setup>
import { ref } from 'vue'
import Child from './Child.vue'

const mensajeHijo = ref('')

const enviar = (texto) => {
    mensajeHijo.value = texto;
}

</script>

<template>
    <div class="container">
        <div class="col-8 offset-2 box-mensaje py-5">
            <h2 class="text-center bg-titulo">Componenten padre</h2>

            <h3 class="text-center">Mensaje recibido: <span class="bg-mensaje px-5 mx-3">{{ mensajeHijo }}</span></h3>
            <Child :mensaje="mensaje" @enviar="enviar" />
        </div>
    </div>
</template>

<style scoped>
.box-mensaje {
    /* From https://css.glass */
    background: rgba(255, 255, 255, 0.2);
    border-radius: 16px;
    box-shadow: 0 4px 30px rgba(0, 0, 0, 0.1);
    backdrop-filter: blur(5px);
    -webkit-backdrop-filter: blur(5px);
    border: 1px solid rgba(255, 255, 255, 0.3);
}

.bg-mensaje {
    background-color: #f0f0f0;
    border-radius: 5px;
}
</style>