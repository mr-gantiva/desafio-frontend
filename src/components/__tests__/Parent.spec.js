import { shallowMount } from '@vue/test-utils'
import Parent from '@/components/Parent.vue'
import { describe, it, expect, test } from 'vitest'

describe('Componente Parent.vue', () => {
  test('Snapshot de Parent.vue', () => {
    // Selección del sujeto de pruebas
    const wrapper = shallowMount(Parent)
    // Validación de match con el snapshot
    expect(wrapper.html()).toMatchSnapshot()
  })
})
