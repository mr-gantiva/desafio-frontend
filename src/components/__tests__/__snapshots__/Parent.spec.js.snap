// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Componente Child.vue > Snapshot de Child.vue 1`] = `
"<div class="container">
  <h2 class="text-center">Componenten hijo</h2>
  <div class="row">
    <div class="col-8 offset-2">
      <div class="container">
        <div class="mb-3"><input type="text" class="form-control" placeholder="Escribe un mensaje"></div>
        <div class="mb-3 d-flex justify-content-center"><button class="btn btn-success">Enviar mensaje al padre</button></div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`Componente Parent.vue > Snapshot de Parent.vue 1`] = `
"<div data-v-91bb33a1="" class="container">
  <div data-v-91bb33a1="" class="col-8 offset-2 box-mensaje py-5">
    <h2 data-v-91bb33a1="" class="text-center bg-titulo">Componenten padre</h2>
    <h3 data-v-91bb33a1="" class="text-center">Mensaje recibido: <span data-v-91bb33a1="" class="bg-mensaje px-5 mx-3"></span></h3>
    <child-stub data-v-91bb33a1=""></child-stub>
  </div>
</div>"
`;
